flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -41,16 +42,49 @@
   private final FileScanTaskReader<T> fileScanTaskReader;
 
   private final InputFilesDecryptor inputFilesDecryptor;
-  private Iterator<FileScanTask> tasks;
+  private final CombinedScanTask combinedTask;
+
+  private Iterator<FileScanTask> fileTasksIterator;
   private CloseableIterator<T> currentIterator;
+  private int fileOffset;
+  private long recordOffset;
 
   public DataIterator(FileScanTaskReader<T> fileScanTaskReader, CombinedScanTask task,
                       FileIO io, EncryptionManager encryption) {
     this.fileScanTaskReader = fileScanTaskReader;
 
     this.inputFilesDecryptor = new InputFilesDecryptor(task, io, encryption);
-    this.tasks = task.files().iterator();
+    this.combinedTask = task;
+
+    this.fileTasksIterator = task.files().iterator();
     this.currentIterator = CloseableIterator.empty();
+
+    // fileOffset starts at -1 because we started
+    // from an empty iterator that is not from the split files.
+    this.fileOffset = -1;
+    this.recordOffset = 0L;
+  }
+
+  public void seek(int startingFileOffset, long startingRecordOffset) {
```

Comment:
This implementation doesn't actually seek to the offsets, it seeks that number of files and/or records. In other words, this only works if it is called before the first call to `next`.

I think that this needs to validate that `fileOffset` is still `-1`. If it isn't, then this should fail immediately with an `IllegalStateException` because seek is only supported just after initialization, not when any records have been consumed. (Or else we'll have to fix this implementation.)

Reply:
added the validation of -1